# Measures of Central Tendency

In [53]:
import pandas as pd

In [54]:
Data = {'ProductA': [25,30,28,32,26,29,27,33,31,28,30,29], 'ProductB': [40,35,38,41,37,36,39,42,38,37,40,36]}

df = pd.DataFrame(Data)
print(df)

    ProductA  ProductB
0         25        40
1         30        35
2         28        38
3         32        41
4         26        37
5         29        36
6         27        39
7         33        42
8         31        38
9         28        37
10        30        40
11        29        36


In [55]:
df.mean()

ProductA    29.00
ProductB    38.25
dtype: float64

### ProductB has a higher average monthly sales.

In [56]:
df.median() 

ProductA    29.0
ProductB    38.0
dtype: float64

### ProductB has a higher median, indicating that the sales distribution tends to be higher on average.

In [57]:
df.mode()

ProductA  ProductB
0      28.0        36
1      29.0        37
2      30.0        38
3       NaN        40

## Significance of the mode:
### ProductA has multiple modes, indicating that there are multiple values occurring with the highest frequency.
### ProductB has multiple modes, indicating that there are multiple values occurring with the highest frequency.

In [58]:
WA = (df['ProductA'] * (1 / len(df))).sum()
WB = (df['ProductB'] * (1 / len(df))).sum()

print(WA)
print(WB)

28.999999999999996
38.25


## Implications of using weighted mean:
### Assuming each month has an equal weight implies that each month contributes equally to the overall mean.
### In this context, the weighted mean provides a simple average that does not give more importance to certain months.

# Measures of Dispersion

In [59]:
RA = df['ProductA'].max() - df['ProductA'].min()
RB = df['ProductB'].max() - df['ProductB'].min()

print(RA)
print(RB)

8
7


## Discussion on variability:

### A larger range suggests greater variability or dispersion in sales, indicating potential fluctuations or outliers.
### A smaller range suggests less variability, implying more consistent sales figures throughout the period.

In [60]:
df.var()

ProductA    5.636364
ProductB    4.750000
dtype: float64

## Discussion on spread of sales:
### A higher variance indicates greater dispersion or variability in sales values.
### A lower variance suggests less spread, indicating more consistency in sales.

In [61]:
df.std()

ProductA    2.374103
ProductB    2.179449
dtype: float64

## Discussion on variability:
### A higher standard deviation indicates greater variability or spread in sales values.
### A lower standard deviation suggests less variability, indicating more consistency in sales.

# Relative Measures of Dispersion

In [62]:
meanA = df['ProductA'].mean()
stdA = df['ProductA'].std()
cvA = (stdA / meanA) * 100

meanB = df['ProductB'].mean()
stdB = df['ProductB'].std()
cvB = (stdB / meanB) * 100

print(cvA)
print(cvB)

8.186561038997239
5.697907115739443


## Discussion on comparing variability:
### The coefficient of variation provides a normalized measure of dispersion, allowing for comparison between products with different average sales.
### A higher coefficient of variation indicates higher relative variability, while a lower coefficient of variation suggests lower relative variability.

In [63]:
rrB = (RB / meanB) * 100

print(rrB)

18.30065359477124


## Discussion on relative range:
### The relative range is a normalized measure of dispersion that accounts for the scale of the data.
### By dividing the range by the mean and expressing it as a percentage, the relative range provides a relative measure of how spread out the data is.
### This is useful for comparing the variability of ProductB in a way that considers the scale of its sales values.

In [64]:
rsdA = cvA

print(rsdA)

8.186561038997239


## Interpretation and implications:
### The relative standard deviation expresses the standard deviation as a percentage of the mean.
### A relative standard deviation of 8.19% for ProductA suggests that the sales values are spread out by this percentage relative to the mean.
### This measure provides a normalized way of understanding the variability of ProductA, accounting for the scale of the data.